### DISCUSSION TITLE
Binary Contents, How to set no Binary compare limit

### DISCUSSION THREAD

**Discussion by petro604** [WinMerge/winmerge#792] [Q&A] [answered]
Under Compare Folder, I am selecting "Binary Contents" to verify a backup.
The default "Binary compare limit" is 64MB, and the maximum it accepts is 2000MB.
How can I set unlimited size?  I want to compare absolutely all the files entirely.
I tried setting the limit to 0MB and it accepts the setting but I'm not sure if that means unlimited.

Can anyone clarify how to compare the entire files?
Using WinMerge 2.16.12.0 x64

**Comment by sdottaka** [comment, marked as answer]
The phrase "Binary compare limit" seems confusing.

"Binary compare limit" is used when the folder comparison method is "Full Contents". The "Binary Contents" method does not use that value.

When the folder comparison is performed by the Full Contents method, the file size exceeds the "Binary compare limit", it is switched to the Binary Contents method and compared.

In the Binary Contents method, if the file sizes do not match, the contents of the files are not compared at all and it is determined that there is a difference.
Also, if the file sizes match, it compares until a mismatch is found in the files.
If the contents of the file match exactly, then the entire file will be compared.

**Reply by Gitoffthelawn** [reply]
@sdottaka Sounds like a UI change is needed to improve clarity.  @petro604 's (mis)understanding sounds reasonable given the current UI.

**Reply by petro604** [reply]
I agree the UI could be made clearer.  But what really confused me was the Help section for that option.  It literally states that the "Binary compare limit" applies to the the "Binary Contents" compare method.

_4.9. Binary compare limit (MB) 
Disabled: Is not active unless the Compare method is set to Binary Contents.
Enabled: Is active when the Compare method is set to Binary Contents._

It may take a little thinking as to how to clarify the UI.  But I would suggest correcting the relevant Help section is something that can be done right away. @sdottaka 's reply was very helpful and IMO could be added to the Help section almost verbatim.  Perhaps something like:

- Disabled: Is not active unless the Compare method is set to Full Contents.
- Enabled: Is active when the Compare method is set to Full Contents.
When the folder comparison is performed by the Full Contents method, if the file size exceeds the "Binary compare limit", it is switched to the Binary Contents method and compared.
For instance if the Binary compare limit is set to 64MB (default), all files up to 64MB will be compared using Full Contents, and any files larger than 64MB will be compared using the Binary Compare method.
The Binary compare limit may be set to any size from 0 to 2000MB.  Setting the Binary compare limit to 0MB will cause Full Compare to [insert whatever will happen. Use Binary compare for ALL files?]

**Reply by Gitoffthelawn** [reply]
@petro604 Thank you for the further details.  I agree.
